### DISCUSSION TITLE
How do we customize syntax color?

### DISCUSSION THREAD

**Discussion by trusktr** [eclipse-langium/langium#1082] [Q&A] [answered]
Is there an example?

**Comment by spoenemann** [comment, marked as answer]
Colors are not determined by the language (server), but by the editor (client):
https://code.visualstudio.com/api/extension-guides/color-theme#syntax-colors

**Reply by msujew** [reply]
In addition, there's also the semantic highlighting service (see [example here](https://github.com/langium/langium/blob/main/packages/langium/src/grammar/lsp/grammar-semantic-tokens.ts)). It's useful in case your textMate highlighting isn't accurate enough/needs more semantic information.

**Reply by trusktr** [reply]
Ah, interesting, so

- if I figure out how to plug my LSP as a TS LSP plugin in https://github.com/langium/langium/discussions/981, then in that case the existing TypeScript colors should play well (easy path),
- otherwise looks like extending `AbstractSemanticTokenProvider` will do the trick (more work).

Thanks!
